### DISCUSSION TITLE
Access Vue instance within Vuex state in Vue3? How to pass vue into vuex?

### DISCUSSION THREAD

**Discussion by apavel3458** [quasarframework/quasar#13189] [General - Components / Directives / etc] [answered]
Hi everyone!
I recently upgraded from quasar 2 to 3, and was able to follow all the helpful docs!  Thanks!  However, I can't seem to resolve this:
 
In Vue2 version of quasar, I accessed Vue global variables as well as router from within Vuex actions using this: 
```
import Vue from "vue"
...
Vue.$axios
```
etc....
However, it appears that in Vue3 and Vuex4, they removed `import Vue from "vue"`.  I was able to access global variables by adding this to my axios boot file:
```
export default boot(({ app, store }) => {
  let axiosInstance = getAxiosInstance(store.state.security.token)
  app.config.globalProperties.$axios = axiosInstance
  store.$axios = axiosInstance
```
So I can access axios from store actions by doing this.$axios.post(...), however now I have two instances of axios, which sucks :(

However, I still need to access the Vue instance within actions to execute things like to make my changing of variables reactive (i know it's not best practice, but I do this too much to fix this right now). 
I tried store.$vue = app in my boot file, but it doesn't work. 

Reading online, they suggest that I pass the app instance into createStore, to have it available, but looks like quasar is creating the store for me. I define it like this: in store/index.js
```
export default function () {
 const store = createStore({
    modules: {
....
    },
    plugins: [createPersistedState()]
  })
```
Any suggestions about how to pass in the Vue instance into the createStore to make it available within store actions (or any other suggestions) would be highly appreciated.   Thanks team!

**Comment by metalsadman** [comment, marked as answer]
use `function` in your vuex actions, not the arrow `=>` form, or you can just export your axios instance, so you can just import it where ever you need it. https://quasar.dev/quasar-cli-vite/ajax-requests#introduction

**Comment by JochemVanIterson** [comment]
Hi!

I think I'm in a similar situation.  In some of my projects I need to access functions and singletons in the `globalProperties` from the app instance, or using provide and inject from the Vue3 package. In the previous version of quasar (and vue) I could add these to the Vue.prototype (I know, not the prettiest solution, but it worked at the time) and access it across the app.

In normal components and from the boot files I am able to get the globalProperties, but I couldn't figure out how to do so inside of the Vuex store (`src/store/index.js`) and it's submodules.

It turns out that the app instance currently isn't passed to the `createStore` function in the `.quasar/app.js` file. Is there a reason why it's implemented this way?

I need to access these properties from getters, so the solution for the vuex actions isn't suited for me (though it might be part of the solution). I also can't export the singletons because they need to be initialized inside of the boot file function.
